apis/v1alpha2/httproute_types.go

```diff
@@ -408,6 +408,25 @@ type HTTPQueryParamMatch struct {
 	Value string `json:"value"`
 }
 
+// HTTPMethod describes how to select a HTTP route by matching the HTTP
+// method as defined by [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-4) and [RFC 5789](https://datatracker.ietf.org/doc/html/rfc5789#section-2).
+// The value is expected in upper case
+// +kubebuilder:validation:Enum=GET;HEADER;POST;PUT;DELETE;CONNECT;OPTIONS;TRACE;PATCH
+type HTTPMethod string
+
+const (
+	HTTPMethodUndefined HTTPMethod = "NONE"
```

Comment:
Do we need this for anything? It does not look like it's a valid value for the field.

Reply:
It is not a valid HTTP method, if that's what you mean. That is the reason I didn't add it to the validator. 

I use such values `not_defined` values in my code and as `genproto` adds one too.  
Should you have reservations against it then I will remove it.

Reply:
Although I think an undefined type like this can make sense in many cases, we've generally avoided them in k8s APIs unless they're going to be explicitly specified. I'd lean towards removing this.